### PR TITLE
Vindarel/legit rebase interactive

### PR DESCRIPTION
**Pull request by vindarel**
Works for the base case, we can pickup/fixup commits, other verbs that need another editing action (edit, reword) don't work.

I wrote up todos and other obvious missing features.

To start a rebase: put the point on a commit, press `r i`.

Awesome though of course :sunglasses:

### CODE REVIEW

File: src/ext/legit/legit.lisp [hunk 1]
```diff
@@ -291,14 +295,21 @@ Next:
...
   Typicaly used to run an external process in the context of a diff buffer command."
   (multiple-value-bind (output error-output exit-code)
       (funcall fn)
-    (declare (ignorable output))
     (cond
       ((zerop exit-code)
-       (message (str:join #\newline (remove-if #'null (list message output)))))
+       (let ((msg (str:join #\newline (remove-if #'null (list message output)))))
+         (when (str:non-blank-string-p msg)
+           (message msg))))
```
Comment (cxxxr) [inline src/ext/legit/legit.lisp:312]: It probably malfunctions if a `~` is included.

**Comment by cxxxr** [comment]
Thank you for the very nice PR!

File: src/ext/legit/legit.lisp (hunk 1 above)
Comment (vindarel) [reply src/ext/legit/legit.lisp:312]: mmh right, but the messages come from porcelain, so it should be fine?

Comment (cxxxr) [reply src/ext/legit/legit.lisp:312]: Hmm, I think it's better to write in a safe way to avoid unnecessary worry.

Comment (cxxxr) [reply src/ext/legit/legit.lisp:312]: That said, it's not enough of a problem to hinder development.
I am very grateful to you for doing such a great job.

**Pull request merged**